### ISSUE TITLE
Annot.get_text("words") - doesn't return the first line of words

### ISSUE THREAD

**Issue by dsclee1** [pymupdf/PyMuPDF#2270]
_**Please provide all mandatory information!**_

## Describe the bug (mandatory)
After getting "TEXT" page annotations I can't get words from the first line of text in the textbox using the .get_text("words") method. It looks as if the bounding rect for the textbox is reading a couple of pixels too far down the page and is missing the first line of the textbox? I could of made a mistake... but I could do with some guidance?

## To Reproduce (mandatory)

```
import fitz
with fitz.open("multiline textbox hell.pdf") as document:
    for page_number, page in enumerate(document):
        words = page.get_text("words")
        textBoxes = []
        for textBox in page.annots(types=(fitz.PDF_ANNOT_FREE_TEXT,fitz.PDF_ANNOT_TEXT)):
            textBoxes.append({ "rect" : textBox.rect, "info" : textBox.info, "words" : textBox.get_text("words")})
print("First word in textbox: 'cover' - x0: " + str(words[15][0]) +" y0: "+str(words[15][1]))
print("Textbox location: - x0: " + str(textBoxes[0]['rect'].x0) + " y0: " + str(textBoxes[0]['rect'].y0))
```

## Your configuration (mandatory)
 - 5.10.0-20-amd64 #1 SMP Debian 5.10.158-2 (2022-12-13) x86_64 GNU/Linux
 - Python 3.9.2
 - PyMuPDF version 1.21.1

## Additional context (optional)
I'm trying to get the content of textboxes that have been placed over the top of existing text, and not read text behind the textbox. My method at the moment is to find the bounding rect of the textbox and remove any text from behind the rect shape. Any guidance on if that is the right approach, or if there's a better method would be great?
[multiline textbox hell.pdf](https://github.com/pymupdf/PyMuPDF/files/10841729/multiline.textbox.hell.pdf)

**Comment by dsclee1** [comment]
The output from the example code:

```
First word in textbox: 'cover' -x0: 71.802001953125 y0: 83.93902587890625
Textbox location: - x0: 71.80169677734375 y0: 86.79302978515625
```

The y0 coords don't look right to me? Shouldn't the word "cover" be contained within the textbox?

**Comment by JorjMcKie** [comment]
> The output from the example code:
> 
> ```
> First word in textbox: 'cover' -x0: 71.802001953125 y0: 83.93902587890625
> Textbox location: - x0: 71.80169677734375 y0: 86.79302978515625
> ```
> 
> The y0 coords don't look right to me? Shouldn't the word "cover" be contained within the textbox?

As this is a FreeText annot, the coordinates don't count here anyway. The text returned for this annotation type is the content of this dictionary value `Annot.info["content"]`.
In PyMuPDF however, I simply call the same MuPDF functions as for normal (page) text ... which ***should*** work.
Presumably things go awkward here, because there is no newline between "cover" and "up", insted a carriage return `\r` only. Between the different operating systems, the only two alternatives for newline are "CRLF" (Windows) and "LF" (oitherwise). CR alone really goes back to the begin of the line ... and thus overwrites "cover".
But if you simply take `Annot.info["content"]` you will get the complete answer.

**Comment by dsclee1** [comment]
Thanks for the quick reply. It makes sense to take the value from `Annot.info["content"]`. Unfortunately that does still cause issues for me in situations where textboxes are double stacked (I have times where customers will amend addresses by continuingly adding textboxes over other textboxes to amend say house numbers and streets) and I need to take the text from only the top textbox (I've attached an example with a textbox stacked over another textbox [multiline textbox hell 2.pdf](https://github.com/pymupdf/PyMuPDF/files/10842636/multiline.textbox.hell.2.pdf)). As the `Annot.info["content"]` doesn't contain the coordinates for each word I can't calculate which words are masked by the next textbox stacked on top.

Basically I'm seeking to only get _visible_ text. Is there a simple way to do that?

Interesting that only carriage returns seem to be present. The textboxes were added using the latest version of Acrobat Reader (2022.003.20322) on a Windows 11 machine. Would have thought it should correctly put in CRLF's?

**Comment by JorjMcKie** [comment]
A PDF stores annotations for each page in an array in the sequence LILO: the last annotation in that array is the most recent one added. This sequence is also returned if you iterate over `page.annots()` or over the xref numbers pertaining to annotations: `page.get_annot_xrefs()`.
So you should be able to find and tell apart anything from each other.

**Comment by JorjMcKie** [comment]
> Interesting that only carriage returns seem to be present. The textboxes were added using the latest version of Acrobat Reader (2022.003.20322) on a Windows 11 machine. Would have thought it should correctly put in CRLF's?

There is a general lack of specification detail when it comes to annotations. So viewers differ in fucntionality (support of border dashing options and border line variants, etc., etc.).

**Comment by dsclee1** [comment]
> A PDF stores annotations for each page in an array in the sequence LILO: the last annotation in that array is the most recent one added. This sequence is also returned if you iterate over `page.annots()` or over the xref numbers pertaining to annotations: `page.get_annot_xrefs()`. So you should be able to find and tell apart anything from each other.

Thanks, that's very helpful. So I can always get the latest amendment as the last item in the array.

Still not sure how I can get around situations where the the latest amendment only covers a small section of the previous amendment though? I will still need to use some text from the previous amendment, but only mask certain words which are replaced by the current amendment. But I don't have word coords?

**Comment by JorjMcKie** [comment]
If that fails in the same ways, as a last resort you can try converting the PDF to a PDF where all annotations have been converted to standard text:
```python
pdfbytes = doc.convert_to_pdf()
doc2 = fitz.open("pdf", pdfbytes)  # an intermediate, converted version.
page=doc[0]
words = page.get_text("words")  # which is the normal full page list of words
```
Here you will find all annot text as the last items with full position info per word.

**Comment by dsclee1** [comment]
Sorry to reopen this, but I'm still having issues. Reading the documentation around Annots I should be able to use the "get_text()" and "get_textbox(rect)" methods.
https://pymupdf.readthedocs.io/en/latest/annot.html#Annot.get_text
https://pymupdf.readthedocs.io/en/latest/annot.html#Annot.get_textbox

I've created a new example. This time with a completely blank doc and only one textbox annotation.
[blank with 1 annotation.pdf](https://github.com/pymupdf/PyMuPDF/files/10848867/blank.with.1.annotation.pdf)

```
import fitz
with fitz.open("blank with 1 annotation.pdf") as document:
    for page_number, page in enumerate(document):
        for textBox in page.annots(types=(fitz.PDF_ANNOT_FREE_TEXT,fitz.PDF_ANNOT_TEXT)):
            print("textBox.type :", textBox.type)
            print("textBox.get_text('words') : ", textBox.get_text('words'))
            print("textBox.get_text('text') : ", textBox.get_text('text'))
            print("textBox.get_textbox() : ", textBox.get_textbox(textBox.rect))
            print("textBox.info['content'] : ", textBox.info['content'])
```

Result:
```
textBox.type : (2, 'FreeText')
textBox.get_text('words') :  []
textBox.get_text('text') :
textBox.get_textbox() :
textBox.info['content'] :  abc123
```

None of the get_text functions seem to be working for my annotation.

**Comment by julian-smith-artifex-com** [comment]
With our latest tree (not yet pushed to github), the output for your new example is:

```
textBox.type : (2, 'FreeText')
textBox.get_text('words') :  [(3.184000015258789, 1.1180419921875, 35.984004974365234, 14.858041763305664, 'abc123', 0, 0, 0)]
textBox.get_text('text') :  abc123

textBox.get_textbox() :  abc123
textBox.info['content'] :  abc123
```

Is this the expected output?

@JorjMcKie may be able to explain why our latest tree is behaving differently from the current release 1.21.1.

We'll be pushing the tree to github in the next few days, and making a new release soon.

**Comment by dsclee1** [comment]
> With our latest tree (not yet pushed to github), the output for your new example is:
> 
> ```
> textBox.type : (2, 'FreeText')
> textBox.get_text('words') :  [(3.184000015258789, 1.1180419921875, 35.984004974365234, 14.858041763305664, 'abc123', 0, 0, 0)]
> textBox.get_text('text') :  abc123
> 
> textBox.get_textbox() :  abc123
> textBox.info['content'] :  abc123
> ```
> 
> Is this the expected output?
> 
> @JorjMcKie may be able to explain why our latest tree is behaving differently from the current release 1.21.1.
> 
> We'll be pushing the tree to github in the next few days, and making a new release soon.

That's what I'm looking for! Perfect. I'll close when the new tree arrives.
